token/program-2022/src/extension/scaled_ui_amount/instruction.rs

```diff
@@ -0,0 +1,139 @@
+#[cfg(feature = "serde-traits")]
+use serde::{Deserialize, Serialize};
+use {
+    crate::{
+        check_program_account,
+        extension::scaled_ui_amount::{PodF64, UnixTimestamp},
+        instruction::{encode_instruction, TokenInstruction},
+    },
+    bytemuck::{Pod, Zeroable},
+    num_enum::{IntoPrimitive, TryFromPrimitive},
+    solana_program::{
+        instruction::{AccountMeta, Instruction},
+        program_error::ProgramError,
+        pubkey::Pubkey,
+    },
+    spl_pod::optional_keys::OptionalNonZeroPubkey,
+    std::convert::TryInto,
+};
+
+/// Interesting-bearing mint extension instructions
+#[cfg_attr(feature = "serde-traits", derive(Serialize, Deserialize))]
+#[cfg_attr(feature = "serde-traits", serde(rename_all = "camelCase"))]
+#[derive(Clone, Copy, Debug, PartialEq, IntoPrimitive, TryFromPrimitive)]
+#[repr(u8)]
+pub enum ScaledUiAmountMintInstruction {
+    /// Initialize a new mint with scaled UI amounts.
+    ///
+    /// Fails if the mint has already been initialized, so must be called before
+    /// `InitializeMint`.
+    ///
+    /// Fails with any number less than 0.
```

Comment:
Correct me if I am wrong, but 0 multiplier is also not allowed, right? Should we also make "number" be a little more specific like "multiplier"?

Reply:
Correct, I initially did allow 0, but then I learned about subnormal floats https://en.wikipedia.org/wiki/Subnormal_number -- I'll fix up this documentation